### PR TITLE
updating cristin contributors when updating existing publication + more

### PR DESCRIPTION
**Pull request by truhacevkir**
- Updating contributors to existing publication when they differ from incoming publication
- Updating curratingInstitutions when updating contributors
- Updating Event.place values when Place i not null
- Setting MediaContribution.disseminationChannel to journalTitle when MediaContribution.mediaPlaceName is missing.

### CODE REVIEW

**Review by brinxmat — APPROVED** [review]

File: cristin-import/src/test/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumerTest.java [hunk 1]
```diff
@@ -881,6 +929,47 @@ void shouldCreatePagesWithPagesBeginEqualToPagesEndWhenCristinJournalPagesBeginI
         assertThat(pages.getBegin(), is(equalTo(pages.getEnd())));
     }
 
+    @Test
+    void shouldMediaContributionDisseminationChannelAsJournalTitleWhenCristinObjectHasJournalTitle()
```
Comment (joachimjorgensen) [inline cristin-import/src/test/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumerTest.java:933]:
```suggestion
    void shouldSetMediaContributionDisseminationChannelAsJournalTitleWhenCristinObjectHasJournalTitle()
```

File: cristin-import/src/main/java/no/unit/nva/cristin/lambda/PublicationUpdater.java [hunk 2]
```diff
@@ -212,4 +264,10 @@ private static AssociatedArtifactList updatedAssociatedLinks(
         list.addAll(incomingAssociatedLinks);
         return new AssociatedArtifactList(list.stream().distinct().toList());
     }
+
+    private static <T> boolean listShouldBeOverwritten(List<T> oldList, List<T> newList) {
+        return !(oldList.size() == newList.size()
+               && new HashSet<>(oldList).containsAll(newList)
+               && new HashSet<>(newList).containsAll(oldList));
```
Comment (joachimjorgensen) [inline cristin-import/src/main/java/no/unit/nva/cristin/lambda/PublicationUpdater.java:271]: Isn't this enough?

```suggestion
        return !oldList.equals(newList);
```

Comment (truhacevkir) [reply cristin-import/src/main/java/no/unit/nva/cristin/lambda/PublicationUpdater.java:271]: Tror ikke det nei, i det tilfelle må to lister være identiske, men i det her tilfelle antar jeg at de er identiske hvis de inneholder samme entries, uansett rekkefølge.

File: cristin-import/src/main/java/no/unit/nva/cristin/lambda/PublicationUpdater.java [hunk 3]
```diff
@@ -177,26 +196,59 @@ private static PublicationInstance<? extends Pages> getPublicationInstance(Publi
...
+                       .withFormat(existingMediaContribution.getFormat())
+                       .withMedium(existingMediaContribution.getMedium())
+                       .withDisseminationChannel(updateDisseminationChannel(existingMediaContribution, incomingMediaContribution))
+                       .build();
         } else {
             return existingPublicationContext;
         }
     }
 
+    private static String updateDisseminationChannel(MediaContribution existingMediaContribution,
```
Comment (joachimjorgensen) [inline cristin-import/src/main/java/no/unit/nva/cristin/lambda/PublicationUpdater.java:219]: Kind of misleading name when you are not actually updating, but rather returning the "most up to date" one.

```suggestion
    private static String getMostUpToDateDisseminationChannel(MediaContribution existingMediaContribution,
```

File: cristin-import/src/main/java/no/unit/nva/cristin/lambda/PublicationUpdater.java (hunk 2 above)
Comment (joachimjorgensen) [reply cristin-import/src/main/java/no/unit/nva/cristin/lambda/PublicationUpdater.java:271]: Ah, okei 👍

File: cristin-import/src/main/java/no/unit/nva/cristin/lambda/PublicationUpdater.java (hunk 3 above)
Comment (joachimjorgensen) [reply cristin-import/src/main/java/no/unit/nva/cristin/lambda/PublicationUpdater.java:219]: "Most up to date" might be misleading as well... `getCorrect` or `getCurrent` might be better? Anyways, up to you if you want to do something with it

**Pull request merged**